### PR TITLE
Convert Translation to PO

### PR DESCRIPTION
**Pull request by VladislavsPerkanuks**
After this PR we should have a working implementation for PO files, PO -> Translation -> PO

**What is done:**
- Rewritten `Translation->PO` implementation
- Added `PO.Marshal(),` to convert PO struct to `[]byte`
- Deleted all old tests` (PO -> Translation)`
- Merged Convert Tests into one.` PO -> Translation -> PO`. That way we are 100% sure that converting back and forth does not change anything in between. Also, `FromPo` and `ToPo` work as intended. 

**What is not done**
- Plural forms are not preserved l in the transformation, but they are crucial to apps that use PO. We need to think of a way to preserve that.
- Renaming all `pot` references to `po`, packages, references, files, ..., as it is more correct

**Note**
~Due to renaming + making changes in the file, git does not think it is the same file (old: pot.go, pot_test.go new: po.go po_test.go), hence the diff is quite big~
**Revert to original pkg and file names (pot) as git, does not like it. (git diff instead renamed+changes, says new file added, old file deleted, which increased PR diff size dramasticly)**

### CODE REVIEW

File: pkg/pot/parse.go [hunk 1]
```diff
@@ -35,11 +65,75 @@ type MessageNode struct {
...
+	return sb.Bytes()
+}
+
 type Po struct {
 	Header   HeaderNode
 	Messages []MessageNode
 }
 
+// Marshal serializes the Po object into a byte slice.
+func (p *Po) Marshal() []byte {
```
Comment (klavinsjanis) [inline pkg/pot/parse.go:125]: Shouldn't Matshal return an error as well? Idiomatic for golang.

Comment (VladislavsPerkanuks) [reply pkg/pot/parse.go:125]: nothing to error there

File: pkg/server/convert.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ func TranslationToData(schema translatev1.Schema, translation *model.Translation
 	case translatev1.Schema_JSON_NGX_TRANSLATE:
 		to = convert.ToNgxTranslate
 	case translatev1.Schema_POT:
```
Comment (janishorsts) [inline pkg/server/convert.go:64]:
```suggestion
	case translatev1.Schema_PO:
```

File: pkg/convert/pot.go [hunk 3]
```diff
@@ -140,11 +133,18 @@ func msgNodeToMF2(node pot.MessageNode, getMessages func(pot.MessageNode) []stri
 		return nil
 	}
 
+	// look for format flag, e.g. #, python-format, c-format, no-python-format, etc.
+	formatFlagIdx := slices.IndexFunc(node.Flags, func(flag string) bool {
+		return strings.Contains(flag, "-format")
```
Comment (janishorsts) [inline pkg/convert/pot.go:138]: I don't know the flag's value, but shouldn't we check the suffix?

```suggestion
		return strings.HasSuffix(flag, "-format")
```

File: pkg/server/convert.go (hunk 2 above)
Comment (VladislavsPerkanuks) [reply pkg/server/convert.go:64]: I reverted back, all changes regarding renaming pot -> po. in pkgs, references, files, etc.
git diff was acting silly.

better to do that in separate PR

File: pkg/convert/pot.go (hunk 3 above)
Comment (VladislavsPerkanuks) [reply pkg/convert/pot.go:138]: you are right

File: pkg/convert/pot.go [hunk 4]
```diff
@@ -236,271 +234,98 @@ func textWithPlaceholders(mfBuilder *mf2.Builder, msg string, placeholders map[s
...
-	if err := write(b, tag, str); err != nil {
-		return fmt.Errorf("write: %w", err)
+	// If original, msgstr are empty
+	if t.Original {
+		patternsToMsg = func(patterns []ast.Pattern) string { return "" }
 	}
 
-	return nil
-}
+	unquoteLiteral := func(l ast.Literal) string { return strings.ReplaceAll(l.String(), "|", "") }
```
Comment (janishorsts) [inline pkg/convert/pot.go:268]: We'll need to resolve how we handle raw and MS2 formatted values. Not part of this PR.

File: pkg/pot/parse.go [hunk 5]
```diff
@@ -16,11 +17,33 @@ type HeaderNode struct {
 	PluralForms PluralForm
 }
 
+func (h *HeaderNode) marshal(b *bytes.Buffer) {
+	b.WriteString("msgid \"\"\nmsgstr \"\"\n")
+
+	if h.Language != language.Und {
+		b.WriteString(fmt.Sprintf("\"Language: %s\\n\"\n", h.Language.String()))
```
Comment (janishorsts) [inline pkg/pot/parse.go:24]: alternative

```suggestion
		b.WriteString(fmt.Sprintf("\"Language: %v\\n\"\n", h.Language))
```

**Review by janishorsts — APPROVED** [review]

File: pkg/pot/parse.go (hunk 5 above)
Comment (VladislavsPerkanuks) [reply pkg/pot/parse.go:24]: yeah, why not

File: pkg/convert/pot.go (hunk 4 above)
Comment (VladislavsPerkanuks) [reply pkg/convert/pot.go:268]: created issue so we wont forget

File: pkg/server/convert.go (hunk 2 above)
Comment (VladislavsPerkanuks) [reply pkg/server/convert.go:64]: created issue #229
